src/FamilyHubs.Referral.Web/Pages/ProfessionalReferral/LocalOfferDetail.cshtml.cs

```diff
@@ -37,35 +37,40 @@ public LocalOfferDetailModel(IOrganisationClientService organisationClientServic
     public async Task<IActionResult> OnGetAsync(string id, string serviceid)
     {
         ServiceId = serviceid;
-        ReturnUrl = Request.Headers["Referer"].ToString();
+        try
+        {
+            ReturnUrl = Request.Headers["Referer"].ToString();
+        }
+        catch
+        {
+            ReturnUrl = "/ProfessionalReferral/LocalOfferResults/BS2%200SP?currentPage=1";

```

Comment:
We can't hardcode a postcode for the services filter page.

There'll be no referer when the user has browsed directly to the service details page, e.g. by visiting a bookmark.

In that case, I don't think we'll have a postcode for the search page, so it probably makes sense to send them back to the postcode search page instead.

Also, instead of using try catch, it'd be simpler to use something like `?.ToString() ?? ""`.

Reply:
?.ToString() can NOT be applied to StringValues also they can NOT be null.